src/Controller/controllers.go

```diff
@@ -105,21 +130,81 @@ func (c Controller) LoadFile(w http.ResponseWriter,
 			errorFlag = true
 		}
 	}
+	w.WriteHeader(200)
 	if errorFlag == true {
 		pad = model.Pad{"", "", errorMessage}
+		w.WriteHeader(500)
 	}
 	jsonAnswer, err := json.Marshal(pad)
-	w.WriteHeader(200)
+
 	fmt.Fprintf(w, "%s", jsonAnswer)
 }
 
+/*
+return the history of pad according to
+pad id
+//TODO: check if file exist in global map
+if not return 500 error
+*/
+func (c Controller) GetPadHistory(w http.ResponseWriter,
+	r *http.Request, p httprouter.Params) {
+
+	w.Header().Set("Access-Control-Allow-Origin", "*")
+	w.Header().Set("Content-Type", "application/json")
+	errorFlag := false
+	//request
+	//take the pad id
+	padRequest := model.PadRequest{}
+	json.NewDecoder(r.Body).Decode(&padRequest)
+	//answer
+	//values of table historyFiles in DB
+	var (
+		id    string
+		state int
+		time  string
+		ip    string
+		//slice with all history values
+		history []PadHistory.PadHistory
+	)
+
+	//TODO check if exist the pad with this id
+
+	//connect to db
+	db, err := sql.Open("mysql", "root:useruseruser@/onlineEditor")
+	if err != nil {
+		errorFlag = true
+	}
+	//close th db
+	defer db.Close()
+
+	//query to db to take the history of pad
+	sqlStatement := `SELECT * FROM historyFiles WHERE id=?`
+	rows, err := db.Query(sqlStatement, padRequest.Id)
+	//iterate the results from query
+	for rows.Next() {
+		//read the values per row
+		err = rows.Scan(&ip, &id, &time, &state)
```

Comment:
Time in db is TIMESTAMP and here it is string, sure you want to do it this way???

Reply:
I think it will be ok.We just want to show it in the page so string i think is fine